src/native/managed/cdacreader/src/Target.cs

```diff
@@ -22,6 +22,15 @@ public struct TargetPointer
     public static implicit operator TargetPointer(ulong v) => new TargetPointer(v);
 }
 
+public struct TargetNUInt
+{
+    public ulong Value;
+    public TargetNUInt(ulong value) => Value = value;
+
+    public static implicit operator ulong(TargetNUInt p) => p.Value;
+    public static implicit operator TargetNUInt(ulong v) => new TargetNUInt(v);
+}
```

Comment:
I wanted something like this for Metadata too, but ended up panicking a bit because it was really tempting to start doing arithmetic on `TargetNUInt`, except that if the target is 32-bit, I couldn't decide if it was ok for things like bitwise negation to touch the upper half or whether it should be clamped to zero.

Reply:
I think we will probably have *some* arithmetic on TargetNUInt, if only some add/subtract logic and maybe a multiply, as that tends to be really useful for dealing with address arithmetic, but arbitrary arithmetic is probably a bit much. I don't think there should be implicit conversion to ulong though. that seems rife for abuse, and mistakes. Implicit conversion FROM ulong seems less risky to me though, that might be ok.